src/api/crypto/blockchain/BalanceOracle.cpp

```diff
@@ -311,30 +294,24 @@ auto BalanceOracle::Imp::process_update_balance(
     Balance input) noexcept -> void
 {
     auto changed{false};
-    auto& data = [&]() -> auto&
-    {
-        if (auto i = data_.find(chain); i != data_.end()) {
-            auto& nymData = i->second.second;
-
-            if (auto j = nymData.find(owner); j != nymData.end()) {
-                auto& out = j->second;
-                changed = (out.first != input);
+    Data* data;
 
-                return out;
-            } else {
-                changed = true;
+    if (auto i = data_.find(chain); i != data_.end()) {
+        auto& nymData = i->second.second;
 
-                return nymData[owner];
-            }
+        if (auto j = nymData.find(owner); j != nymData.end()) {
+            data = &j->second;
```

Comment:
maybe use parentheses, the reader may not be sure about the order of the operators

Reply:
this is not related to order of operators. This is "recent" if contruction where you can declare variable inside of if statement and check some condition on it

Reply:
comment was to line 303, but it was just suggestion

Reply:
ah sorry, my bad then